README.md

```diff
@@ -359,6 +361,34 @@ resources:
 
 To disable the model validation you can use `--disableModelValidation`.
 
+## WebSocket
+
+:warning: *This is an experimental functionality. Please report any bugs or missing features. PRs are welcome.*
+
+Usage in order to send messages back to clients:
+
+`POST http://localhost:{websocketPort}/@connections/{connectionId}`
+
+Or,
+
+```js
+const apiGatewayManagementApi = new AWS.ApiGatewayManagementApi({
+  apiVersion: '2018-11-29',
+  endpoint: event.apiGatewayUrl || `${event.requestContext.domainName}/${event.requestContext.stage}`,
```

Comment:
@computerpunc are you sur this line is correct? Wouldn't the endpoint be `localhost:3001`?

Reply:
Yes, unless you have another idea how to inject address (it can be localhost:3001, localhost:3002, etc.) to the function.

Reply:
So what do we input in the docs:
`endpoint: process.env.IS_OFFLINE ? 'http://localhost:3001' : undefined`
This works well for lambda.invoke.

Reply:
@dherault, @dnalborczyk, @frsechet and others:

Let's start from the beginning:
If AWS would have injected by default `event.requestContext.serverPort`, we wouldn't have this discussion because all information would be at hand.
I could have gone with injecting `event.requestContext.serverPort` but I thought that it was too much to handle all the cases (specific port or default, with http or not) so I went with `apiGatewayUrl`. Any other suitable name will be fine of course.
The drawback, with `process.env.IS_OFFLINE ? 'http://localhost:3001' :` is that the port is hardcoded.
What happens if you want to run 2 local instances of the same app?

So we have the following options:
1. Hardcoded - `http://localhost:3001`.
2. Inject port in `event`- `event.requestContext.serverPort`.
3. Inject port in environment - `process.env.SERVER_PORT`.
4. Use current implementation - `event.apiGatewayUrl`.
5. Use current implementation with renaming `apiGatewayUrl` to something else.

What do you think?

Reply:
on the end of the day it's really up to the user what to use. I personally did use #3 (env, but for the entire endpoint url), but I might switch to this:

`serverless` in the plugin implementation used: https://github.com/serverless/serverless-websockets-plugin/blob/master/example/src/websocket-client.js#L36  as well as https://www.freecodecamp.org/news/real-time-applications-using-websockets-with-aws-api-gateway-and-lambda-a5bb493e9452/

Reply:
@computerpunc  I couldn't find any information on `apiGatewayUrl`. is that something you added to the event? if so, we might want to remove it?

Reply:
@dnalborczyk 
`https://${config.requestContext.domainName}/${config.requestContext.stage}`
is not going to work when running offline. This is what the whole discussion is about. It will not work because the `port` is missing. So the question is how to get the port. Above, I wrote 5 options. Option (1) is my least favorite. Any other is fine. Whatever you find easier to use and more esthetic. No problem to go with hardcoded now (that is remove `apiGarewayUrl`) and add whatever needed support later. 

Since the original PR, `apiGarewayUrl` is injected into the event object (AWS doesn’t do so).

Reply:
ah, thanks. then it should be removed, although helpful, it would be quite confusing to add additional properties on the event, other than the aws provided ones.

Reply:
as far as the docs are concerned, just go with a simple approach, hardcoded. whatever users are deciding to use is their individual choice. (env, hardcoded, etc.)